drivers/gpio/gpio_shell.c

```diff
@@ -21,86 +21,75 @@
 
 LOG_MODULE_REGISTER(gpio_shell);
 
-struct args_index {
-	uint8_t port;
-	uint8_t index;
-	uint8_t mode;
-	uint8_t value;
-};
-
-static const struct args_index args_indx = {
-	.port = 1,
-	.index = 2,
-	.mode = 3,
-	.value = 3,
-};
-
 static int cmd_gpio_conf(const struct shell *sh, size_t argc, char **argv)
 {
 	uint8_t index = 0U;
 	int type = GPIO_OUTPUT;
 	const struct device *dev;
+	int rc = 0;
 
-	if (isdigit((unsigned char)argv[args_indx.index][0]) != 0 &&
-	    isalpha((unsigned char)argv[args_indx.mode][0]) != 0) {
-		index = (uint8_t)atoi(argv[args_indx.index]);
-		if (!strcmp(argv[args_indx.mode], "in")) {
-			type = GPIO_INPUT;
-		} else if (!strcmp(argv[args_indx.mode], "inu")) {
-			type = GPIO_INPUT | GPIO_PULL_UP;
-		} else if (!strcmp(argv[args_indx.mode], "ind")) {
-			type = GPIO_INPUT | GPIO_PULL_DOWN;
-		} else if (!strcmp(argv[args_indx.mode], "out")) {
-			type = GPIO_OUTPUT;
-		} else {
-			return 0;
-		}
+	dev = device_get_binding(argv[1]);
+	if (dev == NULL) {
+		shell_error(sh, "GPIO device %s not available", argv[1]);
+		return -ENODEV;
+	}
+
+	if (!strcmp(argv[3], "in")) {
+		type = GPIO_INPUT;
+	} else if (!strcmp(argv[3], "inu")) {
+		type = GPIO_INPUT | GPIO_PULL_UP;
+	} else if (!strcmp(argv[3], "ind")) {
+		type = GPIO_INPUT | GPIO_PULL_DOWN;
+	} else if (!strcmp(argv[3], "out")) {
+		type = GPIO_OUTPUT;
 	} else {
+		type = shell_strtoul(argv[3], 0, &rc);
+	}
+
+	index = shell_strtoul(argv[2], 10, &rc);
```

Comment:
index is uint8_t, so result of shell_strtoul() needs to be checked for overflow.